### PR TITLE
Adding Snapshot additional mapping tokens

### PR DESCRIPTION
**Pull request by rajrku**
Closes: https://github.com/github/actions-larger-runners/issues/3170

Adding new mapping tokens to snapshot yaml keyword
Added tests to validate the mapping token changes to snapshot keyword

Format:
```
- snapshot:
   - image-name: blah-blah
   - version: <major-version>.*
   - if: <conditions-similar-to-job-if>
  ```

### CODE REVIEW

File: src/Sdk/DTPipelines/workflow-v1.0.json [hunk 1]
```diff
@@ -162,18 +162,35 @@
         "snapshot-mapping"
       ]
     },
-
+
```
Comment (TingluoHuang) [inline src/Sdk/DTPipelines/workflow-v1.0.json:165]: remove the whitespace

File: src/Sdk/DTPipelines/workflow-v1.0.json [hunk 2]
```diff
@@ -162,18 +162,35 @@
...
+          "if": "snapshot-if",
+          "version": {
+            "type": "non-empty-string",
+            "required": false
           }
         }
       }
     },
 
+    "snapshot-if": {
```
Comment (TingluoHuang) [inline src/Sdk/DTPipelines/workflow-v1.0.json:182]: do you need to have 
```
 "always(0,0)",
        "failure(0,0)",
        "cancelled(0,0)",
        "success(0,0)",
```
like other `if` expression?

File: src/Sdk/DTPipelines/Pipelines/ObjectTemplating/PipelineTemplateConverter.cs [hunk 3]
```diff
@@ -376,7 +389,26 @@ internal static Snapshot ConvertToJobSnapshotRequest(TemplateContext context, Te
...
+                Condition = condition,
+                Version = version
+            };
+        }
+
+        private static bool IsSnapshotImageVersionValid(string versionString)
+        {
+            var versionSegments = versionString.Split(".");
+
+            if (versionSegments.Length != 2 ||
```
Comment (TingluoHuang) [inline src/Sdk/DTPipelines/Pipelines/ObjectTemplating/PipelineTemplateConverter.cs:403]: so we only support A.B and not A.B.C for version?
This is different compare to `https://github.com/github/actions-larger-runners/issues/3170`

Comment (rajrku) [reply src/Sdk/DTPipelines/Pipelines/ObjectTemplating/PipelineTemplateConverter.cs:403]: Not yet, we want to start with <major>.* first and get feedback from customers. And then decide if we want to go beyond.

File: src/Sdk/DTPipelines/workflow-v1.0.json (hunk 2 above)
Comment (rajrku) [reply src/Sdk/DTPipelines/workflow-v1.0.json:182]: To start with no. After discussing this with @ericsciple, we want to start small and extend it later. So we chose only to include what we think is sufficient.

**Review by TingluoHuang — APPROVED** [review]

**Pull request merged**